packages/mongodb-rag-ingest/src/commands/embed.ts

```diff
@@ -5,38 +5,59 @@ import {
   withConfig,
   withConfigOptions,
 } from "../withConfig";
-import { updateEmbeddedContent } from "mongodb-rag-core";
+import { logger, updateEmbeddedContent } from "mongodb-rag-core";
 
 type EmbeddedContentCommandArgs = {
-  since: string;
+  since?: string;
   source?: string | string[];
 };
 
 const commandModule: CommandModule<
   unknown,
   LoadConfigArgs & EmbeddedContentCommandArgs
 > = {
-  command: "embed",
+  command: "embed <action>",
+  describe: "Manage embedded content (update or delete)",
   builder(args) {
-    return withConfigOptions(args)
-      .string("since")
-      .option("source", {
-        string: true,
-        description:
-          "A source name to load. If unspecified, loads all sources.",
+    return args
+      .command({
+        command: "update",
+        describe: "Update embedded content data",
+        builder: (updateArgs) =>
+          withConfigOptions(updateArgs)
+            .string("since")
+            .option("source", {
+              string: true,
+              description:
+                "A source name to load. If unspecified, loads all sources.",
+            })
+            .demandOption("since", "Please provide a 'since' date for the update"),
+        handler: ({ since: sinceString, source, ...updateArgs }) => {
+          if (isNaN(Date.parse(sinceString))) {
+            throw new Error(
+              `The value for 'since' (${sinceString}) must be a valid JavaScript date string.`
+            );
+          }
+          const since = new Date(sinceString);
+          withConfig(doEmbedCommand, { ...updateArgs, since, source });
+        },
       })
-      .demandOption("since");
+      .command({
+        command: "delete",
+        describe: "Delete embedded content data",
+        builder: (deleteArgs) =>
+          withConfigOptions(deleteArgs).option("source", {
+            string: true,
+            description:
+              "A source name to delete. If unspecified, deletes all sources.",
+          }),
+        handler: (deleteArgs) => withConfig(doDeleteEmbeddingsCommand, deleteArgs),
+      })
+      .demandCommand(1, "Please specify an action for 'ingest embed' (e.g., 'update' or 'delete')");
   },
-  async handler({ since: sinceString, source, ...args }) {
-    if (isNaN(Date.parse(sinceString))) {
-      throw new Error(
-        `The value for 'since' (${sinceString}) must be a valid JavaScript date string.`
-      );
-    }
-    const since = new Date(sinceString);
-    return withConfig(doEmbedCommand, { ...args, since, source });
+  handler: (args) => {
+    console.error('Specify an action for "embed" command (e.g., "delete" or "update")');
   },
```

Comment:
Its the handler for the base command, if we don't enter either actions. It looks like calling the command without an action like `ingest:pages` is handled by yargs under the hood. I get `Not enough non-option arguments: got 0, need at least 1`. But if I were to call the command with an action that doesn't exist, like `ingest:pages create`, thats when this would trigger.